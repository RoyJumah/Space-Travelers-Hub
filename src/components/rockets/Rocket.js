/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import styles from './css/Rocket.module.scss';
import { makeReservation, cancelReservation } from '../../redux/rockets/rocketsSlice';

const Button = ({ isReserved, id }) => {
  const dispatch = useDispatch();
  if (isReserved) {
    return <button type="button" onClick={() => dispatch(cancelReservation(id))}>Cancel Reservation</button>;
  }
  return <button type="button" onClick={() => dispatch(makeReservation(id))}>Reserve Rocket</button>;
};

const Rocket = (props) => {
  const { rocket } = props;
  const {
    id, name, description, flickr_images, reserved,
  } = rocket;

  return (
    <div className={styles.rocketContainer}>
      <img src={flickr_images} alt="rocket img" className={styles.rocketImg} />
      <div className={styles.col2}>
        <h2>{name}</h2>
        <p>{description}</p>
        <Button id={id} isReserved={reserved} />
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    description: PropTypes.string,
    flickr_images: PropTypes.arrayOf(PropTypes.string),
    reserved: PropTypes.bool,
  }).isRequired,
};

export default Rocket;
