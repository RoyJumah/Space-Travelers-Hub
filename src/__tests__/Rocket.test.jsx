import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Rocket from '../components/rockets/Rocket';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';

const rocket = {id: '1', name: 'falcon1', description: 'description', reserved: false}
describe('Display Rocket component', () => {
  test('it matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rocket rocket = {rocket} />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('should return  Rocket Component', () => {
    render(
      <Provider store={store}>
        <Rocket rocket = {rocket} />
      </Provider>
    );
  });
})
