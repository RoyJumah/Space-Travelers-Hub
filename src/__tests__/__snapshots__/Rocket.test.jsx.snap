// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Rocket component it matches snapshot 1`] = `
<div
  className="rocketContainer"
>
  <img
    alt="rocket img"
    className="rocketImg"
  />
  <div
    className="col2"
  >
    <h2>
      falcon1
    </h2>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <p>
        description
      </p>
    </div>
    <button
      className="reserve__btn"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
